components/experimental/src/units/ratio.rs

```diff
@@ -193,26 +211,194 @@ impl From<u32> for IcuRatio {
+        /// - Mixed fractions: e.g., "1.5/6", "1.4/5.6".
+        /// Note: Scientific notation is not supported.
+        fn parse_fraction(decimal: &str) -> Result<IcuRatio, RatioFromStrError> {
+            let mut components = decimal.split('/');
+            let numerator = components.next();
+            let denominator = components.next();
+            if components.next().is_some() {
+                return Err(RatioFromStrError::MultipleSlashes);
+            }
+
+            let numerator = match numerator {
+                Some(numerator) => parse_decimal(numerator)?,
+                None => IcuRatio::zero(),
+            };
+
+            let denominator = match denominator {
+                Some(denominator) => parse_decimal(denominator)?,
+                None => IcuRatio::one(),
+            };
+
+            if denominator.is_zero() {
+                return Err(RatioFromStrError::DivisionByZero);
+            }
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let mut parts = s.split('/');
-        let numerator = parts.next();
-        let denominator = parts.next();
+            Ok(numerator / denominator)
+        }
+
+        /// Parses a decimal number from a string input.
+        /// Accepts input in various decimal formats, including:
+        /// - Whole numbers: "1"
+        /// - Decimal numbers: "1.5"
+        /// An empty string input is interpreted as "0".
+        /// Note: Fractional inputs are not supported in this context.
+        fn parse_decimal(decimal: &str) -> Result<IcuRatio, RatioFromStrError> {
+            let mut dot_parts = decimal.split('.');
+            let integer_part = dot_parts.next();
+            let decimal_part = dot_parts.next();
+            if dot_parts.next().is_some() {
+                return Err(RatioFromStrError::MultipleDecimalPoints);
+            }
+
+            let integer_part = match integer_part {
+                None | Some("") => IcuRatio::zero(),
+                Some(integer_part) => IcuRatio(
+                    BigInt::from_str(integer_part)
+                        .map_err(RatioFromStrError::ParsingBigIntError)?
+                        .into(),
+                ),
+            };
+
+            let decimal_part = match decimal_part {
+                None | Some("") => return Ok(integer_part),
+                Some(decimal_part) => {
+                    let decimal_power = decimal_part.len() as i32;
+                    let decimal_part = IcuRatio(
+                        BigInt::from_str(decimal_part)
+                            .map_err(RatioFromStrError::ParsingBigIntError)?
+                            .into(),
+                    );
+
+                    decimal_part / IcuRatio::ten().pow(decimal_power) // Divide by 10^decimal_power
+                }
+            };
+
+            Ok(integer_part + decimal_part)
+        }
+
+        let number_str = number_str.replace(',', ""); // Remove commas for parsing
```

Comment:
you can avoid the allocation in the general case:

```suggestion
        let number_str = if number_str.contains(',') {
            Cow::Owned(number_str.replace(',', ""))
        } else {
            Cow::Borrowed(number_str)
        };
```